Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM docker.ocf.berkeley.edu/theocf/debian:stretch
+
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+            git \
+            ruby \
+            ruby-dev \
+            rubygems \
+            build-essential
+
+# stretch only has golang-1.7 as of Dec 2018.
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends -t stretch-backports \
+    golang-1.10
```

Comment:
You shouldn't need to do this in two steps, the stretch container has stretch-backports enabled already and `golang-1.10` is only available in backports as you mentioned, so it'll have to install it from there. I tested this by spinning up a new stretch container and just running the `apt-get install` with `golang-1.10` included in the packages above and that worked just fine.

That being said, this would have to be separate if it didn't have the version in the actual package name like this.